include/DoocsBackendNumericRegisterAccessor.h

```diff
@@ -0,0 +1,371 @@
+
+  template<typename UserType>
+  void DoocsBackendNumericRegisterAccessor<UserType>::doPostRead(TransferType type, bool hasNewData) {
+    DoocsBackendRegisterAccessor<UserType>::doPostRead(type, hasNewData);
+    if(!hasNewData) return;
+
+    // special workaround for D_spectrum: Data type will be DATA_NULL if error is set to "stale data"
+    if(this->dst.type() == DATA_NULL) {
+      // Unfortunately there is no way to get to the data at this point, so fill the buffer with zeros. The data
+      // validity has been set to invalid in this case already by the
+      // DoocsBackendRegisterAccessor<UserType>::doPostRead() call above.
+      return;
+    }
+
+    // verify array length
+    if(size_t(this->dst.length()) < this->nElements + this->elementOffset) {
+      throw ChimeraTK::runtime_error("DoocsBackend: Unexpected array length found in remote property " +
+          this->ea.show_adr() + ": " + std::to_string(this->dst.length()) + " is shorter than the expected " +
+          std::to_string(this->nElements));
+    }
+
+    // define lambda used below for optimisation
+    auto copyFromSourcePointer = [&](const auto* sourcePointer) {
+      using SourceType = std::remove_const_t<std::remove_reference_t<decltype(*sourcePointer)>>;
+      assert(sourcePointer);
+
+      sourcePointer += this->elementOffset;
+      if constexpr(std::is_same<UserType, SourceType>::value) {
+        // raw and target data type match, do a memcopy
+        memcpy(this->buffer_2D[0].data(), sourcePointer, this->nElements * sizeof(SourceType));
+      }
+      else {
+        std::transform(sourcePointer, sourcePointer + this->nElements, this->buffer_2D[0].begin(),
+            [](const SourceType& v) { return ChimeraTK::numericToUserType<UserType>(v); });
+      }
+    };
+
+    // optimise depending on type
+    switch(this->dst.type()) {
+      case DATA_SPECTRUM:
+      case DATA_A_FLOAT: {
+        copyFromSourcePointer(this->dst.get_float_array());
+        return;
+      }
+      case DATA_A_DOUBLE: {
+        copyFromSourcePointer(this->dst.get_double_array());
+        return;
+      }
+      case DATA_A_INT: {
+        copyFromSourcePointer(this->dst.get_int_array());
+        return;
+      }
+      case DATA_A_LONG: {
+        copyFromSourcePointer(this->dst.get_long_array());
+        return;
+      }
+      case DATA_A_BOOL:
+      case DATA_A_SHORT: {
+        copyFromSourcePointer(this->dst.get_short_array());
+        return;
+      }
+      case DATA_IFFF: {
+        for(size_t i = 0; i < this->nElements; i++) {
+          if(i + this->elementOffset == 0) {
+            this->buffer_2D[0][i] = ChimeraTK::numericToUserType<UserType>(dataGet<int>(i + this->elementOffset));
+          }
+          else {
+            this->buffer_2D[0][i] = ChimeraTK::numericToUserType<UserType>(dataGet<float>(i + this->elementOffset));
+          }
```

Comment:
Does an elementOffset make sense with an IFFF? Everything not 0 should not work.

Reply:
I am also wondering why we put everything into one accessor of the same user type. Wasn't there a mechanism to get one int and three floats, and then correlate them with a data consistency group?

Reply:
I removed everything related to IFFF, since this has its own, independent accessor.